modules/library/main/src/main/java/org/geotools/geometry/jts/CircularRing.java

```diff
@@ -173,7 +173,7 @@ public CircularRing normalizeRing() {
         return new CircularRing(ncp, factory, delegate.getTolerance());
     }
 
-    public Geometry reverse() {
+    public Geometry reverseInternal() {
```

Comment:
@aaime this is the "functional" change for the PR.

The superclass now wants to do some book keeping and call reverseInternal() to do the work.

Reply:
reverseInternal is public? Really?

Reply:
probably protected, let me look:  [reverseInternal()](https://github.com/locationtech/jts/blob/master/modules/core/src/main/java/org/locationtech/jts/geom/Geometry.java#L1303)
```
protected abstract Geometry reverseInternal();
```

Reply:
I'd keep it that way then :D